src-plugins/medBinaryOperation/medBinaryOperatorBase.cpp

```diff
@@ -0,0 +1,252 @@
+
+#include "medBinaryOperatorBase.h"
+
+#include <dtkCore/dtkAbstractProcessFactory.h>
+
+#include <itkCastImageFilter.h>
+#include <itkAndImageFilter.h>
+#include <itkOrImageFilter.h>
+#include <itkXorImageFilter.h>
+
+#include <medAbstractDataFactory.h>
+#include <medMetaDataKeys.h>
+#include <medUtilities.h>
+
+
+medBinaryOperatorBase::medBinaryOperatorBase() : medAbstractProcess()
+{
+    m_inputA = NULL;
+    m_inputB = NULL;
+    m_output = NULL;
+}
+
+medBinaryOperatorBase::~medBinaryOperatorBase()
+{
+    m_inputA = NULL;
+    m_inputB = NULL;
+    m_output = NULL;
+}
+
+void medBinaryOperatorBase::setInput ( medAbstractData *data, int channel)
+{
+    if ( channel == 0)
+    {
+        m_inputA = data;
+    }
+
+    if ( channel == 1 )
+    {        
+        m_output = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUChar3" );
+        m_inputB = data;
```

Comment:
Why would setting input B cause the creation of the output instance ? It seems weird to have it here.